src/commands/subscribe.js

```diff
@@ -1,3 +1,18 @@
+import emitMessage from '../commands-utils/emitMessage';
+
 export function subscribe(...args) {
-  return args.length;
+  args.forEach(chan => {
+    if (this.channels.listenerCount(chan) === 0) {
+      this.channels.on(chan, message => {
+        emitMessage(this, chan, message);
+      });
+      // now: client shall enter the subscribed state
```

Comment:
feel free to change these back to TODO, I haven't gotten around to tweak codeclimate yet but it I disagree with many of its default "warnings" 🙂

Reply:
in fact I am working on actually implementing these and to keep the tests (and production code useable), I am thinking to work on Issue #464